js/src/elements/tab-container/element.ts

```diff
@@ -0,0 +1,472 @@
+		const tabListTabWrapper = document.createElement('div');
+		tabListTabWrapper.setAttribute('part', 'tablist-tab-wrapper');
+		tabListTabWrapper.setAttribute('name', 'tablist-tab-wrapper');
+		const tabListSlot = document.createElement('slot');
+		tabListSlot.setAttribute('part', 'tablist');
+		tabListSlot.setAttribute('name', 'tablist');
+		tabListTabWrapper.append(tabListSlot);
+		const panelSlot = document.createElement('slot');
+		panelSlot.setAttribute('part', 'panel');
+		panelSlot.setAttribute('name', 'panel');
+		const beforeTabSlot = document.createElement('slot');
+		beforeTabSlot.setAttribute('part', 'before-tabs');
+		beforeTabSlot.setAttribute('name', 'before-tabs');
+		const afterTabSlot = document.createElement('slot');
+		afterTabSlot.setAttribute('part', 'after-tabs');
+		afterTabSlot.setAttribute('name', 'after-tabs');
+		tabListContainer.append(beforeTabSlot, tabListTabWrapper, afterTabSlot);
+		const afterSlot = document.createElement('slot');
+		afterSlot.setAttribute('part', 'after-panels');
+		afterSlot.setAttribute('name', 'after-panels');
+		shadowRoot.replaceChildren(tabListContainer, panelSlot, afterSlot);
+
+		if (this.#internals && 'role' in this.#internals) {
+			this.#internals.role = 'presentation';
+		} else {
+			this.setAttribute('role', 'presentation');
+		}
+
+		this.addEventListener('keydown', this);
+		this.addEventListener('click', this);
+
+		this.selectTab(-1);
+		this.#setupComplete = true;
+
+		this.initTabContainerChangeListener();
+		this.initTabContainerNavClickListener();
+	}
+
+	attributeChangedCallback(name: string) {
+		if (!this.isConnected || !this.shadowRoot) return;
+		if (name === 'vertical') {
+			this.vertical = this.hasAttribute('vertical');
+		}
+	}
+
+	handleEvent(event: Event) {
+		if (event.type === 'click') return this.#handleClick(event as MouseEvent);
+		if (event.type === 'keydown') return this.#handleKeydown(event as KeyboardEvent);
+	}
+
+	#handleKeydown(event: KeyboardEvent) {
+		const tab = (event.target as HTMLElement)?.closest?.('[role="tab"]');
+		if (!tab) return;
+		const tabs = this.#tabs;
+		if (!tabs.includes(tab as HTMLElement)) return;
+
+		const currentIndex = this.selectedTabIndex;
+		const vertical =
+			tab.closest('[role="tablist"]')?.getAttribute('aria-orientation') === 'vertical';
+		const prevTab = event.code === 'ArrowLeft' || (vertical && event.code === 'ArrowUp');
+		const nextTab = event.code === 'ArrowRight' || (vertical && event.code === 'ArrowDown');
+
+		if (nextTab) {
+			let index = currentIndex + 1;
+			if (index >= tabs.length) index = 0;
+			this.selectTab(index);
+		} else if (prevTab) {
+			let index = currentIndex - 1;
+			if (index < 0) index = tabs.length - 1;
+			this.selectTab(index);
+		} else if (event.code === 'Home') {
+			this.selectTab(0);
+			event.preventDefault();
+		} else if (event.code === 'End') {
+			this.selectTab(tabs.length - 1);
+			event.preventDefault();
+		}
+	}
+
+	#handleClick(event: MouseEvent) {
```

Comment:
`#handleTabClick` to ensure we can differentiate from clicking on the navigation buttons.

Reply:
✅